internal/clients/aws.go

```diff
@@ -162,5 +194,20 @@ func configureNoForkAWSClient(ctx context.Context, ps *terraform.Setup, config *
 	ps.Meta = tfAwsConnsClient
 	fwProvider := xpprovider.GetFrameworkProviderWithMeta(&metaOnlyPrimary{meta: tfAwsConnsClient})
 	ps.FrameworkProvider = fwProvider
+
+	// Register AWS SDK v1 call counter. To learn how SDK v1 session
+	// handler phases map to SDK v2 middleware stack steps, see:
+	// https://aws.github.io/aws-sdk-go-v2/docs/migrating/#handler-phases
+	tfAwsConnsClient.Session.Handlers.Send.PushBack(func(r *awsrequest.Request) {
```

Comment:
Maybe we can also mention that the middleware context of SDK v1 does not contain the service & operation names  so we had to use this approach.